xarray/util/deprecation_helpers.py

```diff
@@ -71,7 +74,7 @@ def func(a, *, b=2):
     licences/SCIKIT_LEARN_LICENSE
     """
 
-    def _decorator(func):
+    def _decorator(func) -> Callable[[T], T]:
```

Comment:
```suggestion
    def _decorator(func: T) -> T:
```
This Typing should go to   `_deprecate_positional_args(version: str) -> Callable[[T], T]:` in line 47. Sorry for the confusion